rar.go

```diff
@@ -37,6 +38,23 @@ func Unrar(source, destination string) error {
 		} else if err != nil {
 			return err
 		}
+		
+		pathComponents := strings.Split(header.Name, "/")
+
+		for pi, path := range pathComponents {
+			// the last component of the path will be the file
+			// so ignore it, since we only want to create folders
+			if pi == len(pathComponents)-1 {
+				continue
+			}
+
+			// check to see if the path exists already
+			if stat, err := os.Stat(destination + path); err != nil || !stat.IsDir() {
+				// make the directory
+				mkdir(destination + path)
```

Comment:
The issue is that header only gets set in the for loop, and this needs to be called after header is set, but before header.isDir is evaluated.